connection_pool.go

```diff
@@ -359,10 +361,20 @@ func (cp *connectionPool) Close() {
 		backend.Close()
 	}
 	cp.backends = nil
+	if !cp.closed {
+		close(cp.closedChan)
```

Comment:
It is safe. The race that Ping encountered was attempting to read from a channel that was being created in another goroutine. That is, the race is actually on the struct that owns the channel object, not the channel itself.